File: prid1920-g13/Models/Tag.cs

using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;

namespace prid_1819_g13.Models
{
    public class Tag
    {
        [Key]
        public int Id { get; set; }
        [Required(ErrorMessage = "Name Required")]
        public string Name { get; set; }
        [NotMapped]
        public int num {get => PostTags.Where(x => x.TagId == Id).ToList().Count();} // {get => PostTags.Where(x => x.TagId == Id).ToList().Count();}
        public virtual List<PostTag> PostTags { get; set; } =new List<PostTag>();
        [NotMapped]
        public IEnumerable<Post> Posts
        {
            get => PostTags.Select(f => f.Post);
        }
    }
}

File: prid1920-g13/ClientApp/src/app/models/Tag.ts
export class Tag {
    name: string;
    id: any;
    num: number;
    
    constructor(data: any) {
      if (data) {
        this.name = data.name;
        this.id = data.id;
        this.num = data.num;
      }
    }
  }
